### DISCUSSION TITLE
How to search multiple messages through an Excel file

### DISCUSSION THREAD

**Discussion by stefano-b** [nextgenhealthcare/connect#4528] [Q&A]
Hi all, 

I have a Mirth Connect Server 3.3.2.7911 with Java Java version: 1.8.0_241 and i need to reprocess more than 1200 messages on a single channel. In order to identify the messages I have an Excel file with some data (for example the patien order ID or surname) but for do that I have to search manually each message.

Using the Advanced Search Filter i tried to insert more Order ID's : 

![Advanced  Search Filter](https://user-images.githubusercontent.com/83481538/116765848-9c708b80-aa27-11eb-995d-03a6e54596f9.png)

Unfortunately the search failed so I need to copy and paste the order ID for each message.

Do you know a smart and quick way for search multiple messages from my Excel file?

Note that I don't have pretty experience with Mirth Connect and any kind of coding.

Many thanks all for your support!

Stefano

**Comment by tom08zehn** [comment]
As far as I know you can search using a regular expression which could hold simply a list of all your order IDs. Of course I assume that this would be a very long reg-ex...

A probably better way is to reprocess messages using a batch file and the Mirth API, here's a simple example that can enable and deploy a channel and reprocess a message with a certain ID. Of course this needs some modification for your purpose...

```bat
@ECHO OFF

rem path to curl.exe
SET CURL="C:\curl.exe"

rem channel id
SET CHID=d7f5ef8c-f7bf-457d-8ba2-2cdca3095392

rem message id to be reprocessed (0 for none)
SET MSGID=0

rem mirth credentials
SET USRPWD=admin:secret

rem --------------------------------------------------------------------------

rem enable channel
echo|set /p="Enabling channel... "
%CURL% -X POST --basic --user %USRPWD% -k https://localhost:8443/api/channels/%CHID%/enabled/true

rem deploy channel
echo|set /p="Deploying channel... "
%CURL% -X POST --basic --user %USRPWD% -k https://localhost:8443/api/channels/%CHID%/_deploy?returnErrors=true

rem reprocess message
if %MSGID%==0 (
    goto end
) else (
    goto reprocess
)

:reprocess
echo|set /p="Reprocessing message %MSGID%... "
%CURL% -X POST --basic --user %USRPWD% -k "https://localhost:8443/api/channels/%CHID%/messages/%MSGID%/_reprocess?replace=true&filterDestinations=false"

:end

echo DONE!
pause
```
